http/query.go

```diff
@@ -259,17 +220,24 @@ func (r QueryRequest) proxyRequest(now func() time.Time) (*query.ProxyRequest, e
 	// Query is preferred over spec
 	var compiler flux.Compiler
 	if r.Query != "" {
-		compiler = lang.FluxCompiler{
-			Query: r.Query,
-		}
-	} else if r.AST != nil {
-		var err error
-		r.Spec, err = toSpec(r.AST, now)
+		pkg, err := flux.Parse(r.Query)
 		if err != nil {
 			return nil, err
 		}
-		compiler = lang.SpecCompiler{
-			Spec: r.Spec,
+		if r.Extern != nil {
+			pkg.Files = append([]*ast.File{r.Extern}, pkg.Files...)
```

Comment:
Since order doesn't matter should we just append this instead of prepending?

Reply:
Do we want to add the extern concept to the ASTCompiler itself, so that the appending of the file is normal and standard?

Reply:
I have been playing around with this and I think it makes sense to leave it how it is. This way its more flexible. In the testing code I am writing I am likely going to append multiple ast.Files so exposing a single Extern ast.File is not that helpful. Exposing a list seems to specific, so I say lets leave it.

Reply:
The interpreter still executes statements sequentially. Dependency reordering has not been implemented yet, which is why I am prepending this file for right now.